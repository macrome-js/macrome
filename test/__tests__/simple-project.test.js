const { resolve } = require('path');
const { writeFile, readFile, unlink } = require('fs').promises;

const { testProject } = require('../test-project');
const { sandboxPath, gitStatus, eventually } = require('../utils');

describe('simple project', () => {
  const projectPath = sandboxPath('projects/simple-project');
  const test = testProject(projectPath);

  describe('watch', () => {
    test.watchSetup();

    it('simple-project.js', async () => {
      const filePath = resolve(projectPath, 'lib/simple-project.js');
      const originalContent = await readFile(filePath, 'utf8');

      await unlink(filePath);

      await eventually(() => {
        expect(gitStatus(projectPath)).toMatchInlineSnapshot(`
          Array [
            " D lib/generated-simple-project.js",
            " D lib/simple-project.js",
          ]
        `);
      });

      await writeFile(filePath, originalContent + '\n');

      await eventually(() => {
        expect(gitStatus(projectPath)).toMatchInlineSnapshot(`
          Array [
            " M lib/generated-simple-project.js",
            " M lib/simple-project.js",
          ]
        `);
      });

      await writeFile(filePath, originalContent);

      await eventually(() => {
        expect(gitStatus(projectPath)).toMatchInlineSnapshot(`Array []`);
      });
    });
  });
});
