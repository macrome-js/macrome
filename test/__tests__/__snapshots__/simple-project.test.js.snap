// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple project builds 1`] = `""`;

exports[`simple project cleans 1`] = `
Array [
  " D lib/generated-simple-project.js",
]
`;
